### PR TITLE
add optional self consistency

### PR DESCRIPTION
**Pull request by depshad**
Added self consistency as a boolean parameter (If we have token budget)
From https://arxiv.org/abs/2407.01476, _We also observe that
self-consistency is essential for the performance of search (28.5% → 37.0%)._ 
Reference: https://github.com/kohjingyu/search-agents/blob/main/agent/value_function.py
I could not find the sampling option in chat agent(same as openai so used a for loop). Can someone share how to do that?

### CODE REVIEW

**Comment by SgtVincent** [comment]
Thanks Ekansh for adding this nice feature. 

@richielo Could you please have a look at this update since this is on the file you created?

File: mcts_action/demo/vf_agent.py [hunk 1]
```diff
@@ -1,89 +1,137 @@
...
+        if self.use_self_consistency:
+            # Self-consistency: Sample multiple reasoning paths
+            all_scores = []
+            for _ in range(self.n_samples):
+                response = self._agent.step(user_msg)
+                score = self._parse_response(response.msgs[0].content)
+                all_scores.append(score)
+
+            # Compute the final score as the mean of all sampled scores
+            return np.mean(all_scores)
```
Comment (richielo) [inline mcts_action/demo/vf_agent.py:133]: Can we get the option to do majority voting too instead of just mean?

**Review by richielo — CHANGES_REQUESTED** [review]

File: mcts_action/demo/vf_agent.py (hunk 1 above)
Comment (depshad) [reply mcts_action/demo/vf_agent.py]: Added the MAJORITY_VOTE

**Review by richielo — APPROVED** [review]

**Comment by SgtVincent** [comment]
LGTM! Thanks guys!

**Pull request merged**